lib/rubocop/cop/rspec/context_wording.rb

```diff
@@ -35,41 +35,71 @@ module RSpec
       #     # ...
       #   end
       #
+      # This cop can be customized allowed context description pattern
+      # with `AllowedPatterns`. By default, there are no checking by pattern.
+      #
+      # @example `AllowedPatterns` configuration
+      #
+      #   # .rubocop.yml
+      #   # RSpec/ContextWording:
+      #   #   AllowedPatterns:
+      #   #     - /とき$/
+      #
+      # @example
+      #   # bad
+      #   context '条件を満たす' do
+      #     # ...
+      #   end
+      #
+      #   # good
+      #   context '条件を満たすとき' do
+      #     # ...
+      #   end
+      #
       class ContextWording < Base
-        MSG = 'Start context description with %<prefixes>s.'
+        include AllowedPattern
+
+        MSG = 'Context description should match %<patterns>s.'
 
         # @!method context_wording(node)
         def_node_matcher :context_wording, <<-PATTERN
-          (block (send #rspec? { :context :shared_context } $(str #bad_prefix?) ...) ...)
+          (block (send #rspec? { :context :shared_context } $(str $_) ...) ...)
         PATTERN
 
         def on_block(node) # rubocop:disable InternalAffairs/NumblockHandler
-          context_wording(node) do |context|
-            add_offense(context,
-                        message: format(MSG, prefixes: joined_prefixes))
+          context_wording(node) do |context, description|
+            if bad_pattern?(description)
+              message = format(MSG, patterns: expect_patterns)
+              add_offense(context, message: message)
+            end
           end
         end
 
         private
 
-        def bad_prefix?(description)
-          !prefix_regex.match?(description)
+        def allowed_patterns
+          @allowed_patterns ||= begin
+            prefix_regexes = prefixes.map { |pre| /^#{Regexp.escape(pre)}\b/ }
+            super.concat(prefix_regexes)
+          end
```

Comment:
I was looking into _why_ the memoization was necessary, and found that `cop_config["AllowedMethods"]` is the same object as `Array(cop_config["AllowedMethods"])`, and that `concat` is a mutating method. And then I found rubocop/rubocop#10953 with your name on it 😄

Reply:
Perhaps we should follow the new pattern and do e.g.
```rb
def allowed_patterns
  super + prefix_regexes
end

def prefix_regexes
  @prefix_regexes ||= begin
    prefixes = Array(cop_config.fetch('Prefixes', []))
    prefixes.map { |pre| /^#{Regexp.escape(pre)}\b/ }
  end
end
```

Reply:
Wow. almost made the same mistake again😅 Thanks for catching that.